### PR TITLE
[EarlyReturn] Add ReturnBinaryOrToEarlyReturnRector

### PR DESCRIPTION
**Pull request by samsonasik**
To Changes Single return of || to early returns

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/early-return/src/Rector/Return_/ReturnBinaryOrToEarlyReturnRector.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Changes Single return of || to early returns', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function accept($something, $somethingelse)
+    {
+        return $something || $somethingelse;
```
Comment (TomasVotruba) [inline rules/early-return/src/Rector/Return_/ReturnBinaryOrToEarlyReturnRector.php:49]: Notes from call - keep simple expressions:

```php
// keep
return 1 || 2 || 5;


if (true && $value) {
}
```

Update method calls and static calls only:

```php
// changed
return $this->process(...) || $this->someMethod(...) || some::statiCall() || true;
```

Comment (samsonasik) [reply rules/early-return/src/Rector/Return_/ReturnBinaryOrToEarlyReturnRector.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you :tada:

**Pull request merged**